File: NET_8/FoodCall.Domain/Exceptions/EntityNotFoundException.cs
namespace FoodCall.Domain.Exceptions;

public class EntityNotFoundException : DomainException
{
    public EntityNotFoundException(string entityName, Guid id) 
        : base($"{entityName} com ID '{id}' não foi encontrado(a).")
    {
    }

    public EntityNotFoundException(string entityName, string identifier) 
        : base($"{entityName} com identificador '{identifier}' não foi encontrado(a).")
    {
    }
}

File: Java/src/main/java/com/foodcall/domain/exceptions/EntityNotFoundException.java
package com.foodcall.domain.exceptions;

import java.util.UUID;

public class EntityNotFoundException extends DomainException {
    
    public EntityNotFoundException(String entityName, UUID id) {
        super(String.format("%s com ID '%s' não foi encontrado(a).", entityName, id));
    }
    
    public EntityNotFoundException(String entityName, String identifier) {
        super(String.format("%s com identificador '%s' não foi encontrado(a).", entityName, identifier));
    }
}